src/accountsdb/account_store.zig

```diff
@@ -64,28 +83,37 @@ pub const AccountReader = union(enum) {
     }
 };
 
-/// Interface for both reading and writing accounts.
-///
-/// Do not use this unless you need to *write* into the database generically.
-/// Otherwise use AccountReader if you only need to read accounts.
-pub const AccountStore = union(enum) {
-    accounts_db: *AccountsDB,
-    thread_safe_map: *ThreadSafeAccountMap,
+/// Interface for only reading accounts that should be visible for a particular slot.
+pub const SlotAccountReader = union(enum) {
+    accounts_db: struct { *AccountsDB, *const Ancestors },
+    thread_safe_map: struct { *ThreadSafeAccountMap, *const Ancestors },
+    simple_map: struct { Allocator, *const std.AutoArrayHashMapUnmanaged(Pubkey, Account) },
```

Comment:
Not sure about this allocator. It looks like get is used on the map directly, which wouldn't duplicate the account's memory (meaning we wouldn't need the allocator on deiniting it).

Reply:
good point. is this better? 873fb8df3e757edb15492930ab6f4acfeaed50d0

Reply:
Yep. Alternatively it could be a no-op allocator? But this is good

Reply:
I made it panic because I'd like be alerted if it's being misused.